src/service/cid-impl.js

```diff
@@ -182,15 +182,15 @@ function getOrCreateCookie(cid, getCidStruct, persistenceConsent) {
   const newCookie = 'amp-' + cid.sha384Base64_(getEntropy(win));
 
   // Store it as a cookie based on the persistence consent.
-  persistenceConsent.then(() => {
+  return persistenceConsent.then(() => {
```

Comment:
This will not work. These promises have a cyclic dep.

The real fix is to use the `baseCid_` instance var to short circuit the second entry into this function without reading cookies.